test/unit/pg.ddl/generateType.js

```diff
@@ -1,114 +1,155 @@
+  } catch (err) {
+    test.fail(err);
+    test.end();
+    return;
+  }
 
-  test.strictSame(ddl.generateType('Integer', ms.domains.get('Integer')), {
+  test.strictSame(ddl.generateType('Integer', schema.domains.get('Integer')), {
     type: 'integer',
   });
 
-  test.strictSame(ddl.generateType('Count', ms.domains.get('Count')), {
-    type: 'double precision',
-  });
-
-  test.strictSame(ddl.generateType('Nomen', ms.domains.get('Nomen')), {
-    type: 'text',
-  });
-
-  test.strictSame(ddl.generateType('Lambda', ms.domains.get('Lambda')), {
-    type: 'text',
-  });
-
-  test.strictSame(ddl.generateType('Logical', ms.domains.get('Logical')), {
-    type: 'boolean',
-  });
-
-  test.strictSame(ddl.generateType('SHA2', ms.domains.get('SHA2')), {
-    type: 'bytea',
-  });
-
-  test.strictSame(ddl.generateType('EnumType', ms.domains.get('EnumType')), {
-    type: '"EnumType"',
-    sql: `
+  test.strictSame(
+    ddl.generateType('EnumType', schema.domains.get('EnumType')),
+    {
+      type: '"EnumType"',
+      sql: `
 -- Enum: "EnumType" ------------------------------------------------------------
 
 CREATE TYPE "EnumType" AS ENUM (
   'value1',
   'value2',
   'value3'
 );`,
-  });
+    }
+  );
 
   test.strictSame(
-    ddl.generateType('FlagsSmallInt', ms.domains.get('FlagsSmallInt')),
+    ddl.generateType('FlagsSmallInt', schema.domains.get('FlagsSmallInt')),
     { type: 'smallint' }
   );
 
-  test.strictSame(ddl.generateType('FlagsInt', ms.domains.get('FlagsInt')), {
-    type: 'integer',
-  });
+  test.strictSame(
+    ddl.generateType('FlagsInt', schema.domains.get('FlagsInt')),
+    {
+      type: 'integer',
+    }
+  );
 
   test.strictSame(
-    ddl.generateType('FlagsBigInt', ms.domains.get('FlagsBigInt')),
+    ddl.generateType('FlagsBigInt', schema.domains.get('FlagsBigInt')),
     { type: 'bigint' }
   );
 
-  test.strictSame(ddl.generateType('Flags16', ms.domains.get('Flags16')), {
+  test.strictSame(ddl.generateType('Flags16', schema.domains.get('Flags16')), {
     type: 'smallint',
   });
 
-  test.strictSame(ddl.generateType('Flags32', ms.domains.get('Flags32')), {
+  test.strictSame(ddl.generateType('Flags32', schema.domains.get('Flags32')), {
     type: 'integer',
   });
 
-  test.strictSame(ddl.generateType('Flags64', ms.domains.get('Flags64')), {
+  test.strictSame(ddl.generateType('Flags64', schema.domains.get('Flags64')), {
+    type: 'bigint',
+  });
+
+  test.throws(
+    () =>
+      ddl.generateType(
+        'FlagsUnsupported',
+        schema.domains.get('FlagsUnsupported')
+      ),
+    new Error('Too many flags in FlagsUnsupported, must not be bigger than 64')
+  );
+
+  test.end();
+});
+
+metatests.test('pg.ddl.generateType unit test(system domains)', async test => {
+  let errors;
+  let schema;
+
+  try {
+    [errors, schema] = await metaschema.fs.load(
+      'schemas/system',
+      options,
+      config
+    );
+    if (errors.length !== 0) test.bailout(errors);
```

Comment:
AFAIK, `test.bailout()` throws, so it should probably be moved outside of the `try` block.